pkg/client/client_test.go

```diff
@@ -51,10 +57,24 @@ var _ = Describe("Client", func() {
 				},
 			},
 		}
+		// Pod is invalid without a container field in the PodSpec
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("pod-%v", count), Namespace: ns},
+			Spec:       corev1.PodSpec{},
+		}
 		node = &corev1.Node{
 			ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("node-name-%v", count)},
 			Spec:       corev1.NodeSpec{},
 		}
+		// Add Deployment and Node to the Scheme.
+		scheme = runtime.NewScheme()
+		dgv := schema.GroupVersion{Group: "apps", Version: "v1"}
+		ngv := schema.GroupVersion{Group: "", Version: "v1"}
+		dgvk := dgv.WithKind("Deployment")
+		ngvk := ngv.WithKind("Node")
+		scheme.AddKnownTypeWithName(dgvk, dep)
+		scheme.AddKnownTypeWithName(ngvk, node)
```

Comment:
`client` users should not be required to register core types, by default, `client` should use scheme from the core.

Reply:
Done.